copilot/core/vectordb_utils.py

```diff
@@ -98,7 +98,9 @@ def index_file(ext, item_path, chroma_client):
 
         # Split the document and add it to the collection
         copilot_debug(f"File with md5 {md5} added to index with 'purge': False.")
-        documents = text_splitter.split_documents([document])
+        documents: list[Document] = [document]
+        if text_splitter:
```

Comment:
text_splitter should always have a value, it would be more efficient to throw an exception if it does not, rather than follow silently.  Later, for Java files returns None, which is not valid.

Reply:
In the demo meeting we had defined some Java parameters and the other code formats for the max chunk size, I have added them.  I have added a warning if there is no splitter for a format, but I would not make it mandatory because some formats do not have a proper splitter.